### PR TITLE
Handle merchant_token_prop in Prop.cc

### PR DESCRIPTION
**Pull request by jez**
### Motivation

We refactored the way that some of Stripe's database layer works (in the service
of modularity).

We need to update Sorbet to match the new logic.

### Commit summary

Review by commit; there are a lot of spurious exp file changes.

- **Move merchant_prop stuff out of prop.rb** (5ed8b9ad1)

  prop.rb is a huge test and I hate it; much rather have stuff in smaller
  files.

  No new behavior, no new tests, just a different file.

- **Add failing test case** (0c65f6b08)

- **Fix failing test** (138ec1f03)

### Test plan

See included automated tests.

### CODE REVIEW

**Comment by jez** [comment]
cc @jnovak-stripe

**Comment by jez** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_K9M1lY83rac3L5
→ https://go/builds/bui_K9M1V8fwUtbSc6

File: rewriter/Prop.cc [hunk 1]
```diff
@@ -152,11 +152,32 @@ optional<PropInfo> parseProp(core::MutableContext ctx, const ast::Send *send) {
...
+            // This is a lie; technically the API allows an optional `merchant:` keyword arg to
+            // customize the method name. It defaults to `:merchant`
             ret.name = core::Names::merchant();
             // 5 is the length of the _prop suffix
             ret.nameLoc = core::LocOffsets{send->loc.beginPos(), send->loc.endPos() - 5};
             ret.type = ast::MK::Constant(send->loc, core::Symbols::String());
             break;
+        case core::Names::merchantTokenProp().rawId():
+            ret.isImmutable = true;
+            // This is a lie; technically the API allows an optional `merchant:` keyword arg to
```
Comment (elliottt) [inline rewriter/Prop.cc:164]:
```suggestion
            // This is a lie; technically the API allows an optional `name:` keyword arg to
```

Comment (elliottt) [reply rewriter/Prop.cc:164]: Will defaulting the name to `:merchant` be problematic for go-to-definition or the compiler?

File: rewriter/Prop.cc [hunk 2]
```diff
@@ -152,11 +152,32 @@ optional<PropInfo> parseProp(core::MutableContext ctx, const ast::Send *send) {
         }
         case core::Names::merchantProp().rawId():
             ret.isImmutable = true;
+            // This is a lie; technically the API allows an optional `merchant:` keyword arg to
```
Comment (elliottt) [inline rewriter/Prop.cc:155]:
```suggestion
            // This is a lie; technically the API allows an optional `name:` keyword arg to
```

**Review by elliottt — APPROVED** [review]
Looks good! My only question was about whether or not defaulting the name to `:merchant` is a problem.

File: rewriter/Prop.cc (hunk 1 above)
Comment (jez) [reply rewriter/Prop.cc:164]: it's definitely not the right thing to do, but it's the same problem that used to exist for `merchant_prop`, and that case hasn't come up yet, so I'm not worried about fixing it in this PR

**Pull request merged**